Description: Upstream fix for SMTP protocol (original
 test issues both EHLO and HELO, regardless of server reply).
Origin: upstream
Bug-Debian: http://bugs.debian.org/614984

---
 protocols/smtp.c |   12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

--- a/protocols/smtp.c
+++ b/protocols/smtp.c
@@ -63,17 +63,11 @@
 int check_smtp(Socket_T s) {
   ASSERT(s);
   
-  if (! expect(s, 220, TRUE))
-    return FALSE;
-  
   /* Try HELO also before giving up as of rfc2821 4.1.1.1 */
-  if (! (say(s, "EHLO localhost\r\n") && expect(s, 250, FALSE)) || ! (say(s, "HELO localhost\r\n") && expect(s, 250, TRUE)))
-    return FALSE;
+  if (expect(s, 220, TRUE) && ((say(s, "EHLO localhost\r\n") && expect(s, 250, FALSE)) || (say(s, "HELO localhost\r\n") && expect(s, 250, TRUE))) && (say(s, "QUIT\r\n") && expect(s, 221, TRUE)))
+    return TRUE;
 
-  if (! (say(s, "QUIT\r\n") && expect(s, 221, TRUE)))
-    return FALSE;
-    
-  return TRUE;
+  return FALSE;
 }
 
 
